lib/src/io.dart

```diff
@@ -867,38 +840,38 @@ String _findTarPath() {
 /// Extracts a `.tar.gz` file from [stream] to [destination].
 Future extractTarGz(Stream<List<int>> stream, String destination) async {
   log.fine('Extracting .tar.gz stream to $destination.');
-  if (Platform.isWindows) {
-    return await _extractTarGzWindows(stream, destination);
-  }
+  final decompressed = stream.transform(GZipCodec().decoder);
+  // We used to stream directly to `tar`,  but that was fragile in certain
+  // settings.
+  PubProcessResult processResult;
+  await withTempDir((tempDir) async {
+    final tarFile =
+        path.join(tempDir, path.basenameWithoutExtension(destination));
```

Comment:
Why not just always call this intermediate file for `path.join(tempDir, 'archive.tar')` ?